src/info/url.rs

```diff
@@ -16,28 +17,16 @@ impl UrlInfo {
     }
 }
 
-pub fn get_repo_url(repo: &Repository, hide_token: bool, http_url: bool) -> String {
-    let config = repo.config_snapshot();
-    let remotes = match config.plumbing().sections_by_name("remote") {
-        Some(sections) => sections,
-        None => return String::default(),
+pub fn get_repo_url(repo: &Repository, hide_token: bool, http_url: bool) -> Result<String> {
+    let remote = match repo.try_find_remote("origin") {
+        Some(remote) => remote?,
+        None => return Ok(String::new()),
     };
 
-    let mut remote_url: Option<String> = None;
-    for (name, url) in remotes.filter_map(|section| {
-        let remote_name = section.header().subsection_name()?;
-        let url = section.value("url")?;
-        (remote_name, url).into()
-    }) {
-        remote_url = url.to_string().into();
-        if name == "origin" {
-            break;
-        }
-    }
-
-    match remote_url {
-        Some(url) => format_url(&url, hide_token, http_url),
-        None => String::default(),
+    if let Some(url) = remote.url(gix::remote::Direction::Push) {
+        Ok(format_url(&url.to_string(), hide_token, http_url))
+    } else {
+        Ok(String::new())
```

Comment:
IDK if you have a preference, but this can also be written with `Option::map` and `Option::unwrap_or_default` I think.

Reply:
done